teps/0019-other-arch-support.md

```diff
@@ -0,0 +1,424 @@
+
+## Proposal
+
+Extend current Tekton CI/CD system to run Tekton tests and builds for non-amd64
+architectures. As a result Tekton artifacts (container images and yaml files)
+will be tested on other hardware architectures and can be released in the same
+way as it is done for amd64.
+
+Architecture specific builds/tests should be done on real hardware (no emulation
+on top of amd64). That means that native hardware(for specific architecture)
+should be provided by interested parties. Tekton lives on top of k8s cluster,
+so native hardware should provide ability to get new k8s cluster on demand or
+at least have preinstalled k8s cluster(s), which can be cleaned/recreated fast.
+
+It makes no sense to setup full CI/CD infrastructure on native hardware, amd64
+Tekton cluster (dogfooding) has already functionality which can be used as main
+server and architecture specific cluster can be used when it is required
+(actual tests or builds).
+amd64 Tekton cluster will take responsibilities:
+- to initiate all actions.
+- to show results/logs in the UI.
+- to keep all secrets and interact with storages, registries, external systems,
+ etc.
+- to interact with google cloud (google sdk is available only for amd64).
+
+Non-amd64 k8s cluster will:
+- operate on native hardware.
+- build the non-amd64 container images.
+- run the tests.
+
+All steps should be packed as Tekton tasks, that allows to use current
+dogfooding cluster (or similar setup) to build non-amd64 Tekton artifacts via
+Tekton.
+
+### Requirements
+
+amd64 Tekton cluster should be able to:
+- communicate with non-amd64 k8s cluster via kubectl or ssh where it's
+  necessary.
+- have access to non-amd64 local container registry.
+- show the results in the UI
+
+Non-amd64 k8s cluster should have:
+- several worker nodes.
+- local container registry.
+- default storage.
+- access to Internet or required container images stored in local registry.
+
+### Design details
+
+To target some part of the logic to native k8s cluster it is enough to specify
+`kubeconfig` with information about access to non-amd64 cluster, for instance
+`go test ... -kubeconfig ...` to run e2e tests, `export KUBECONFIG ...` and
+`kubectl delete/get ...` for simple kubectl cleanup/setup tasks or `kubectl 
+apply -f *.yaml` for more complex actions.
+Current tasks available at [catalog](https://github.com/tektoncd/catalog/tree/master/task)
+can be used as is ([kubernetes-actions](https://github.com/tektoncd/catalog/tree/master/task/kubernetes-actions/0.1))
+or extended ([kubectl-deploy-pod](https://github.com/tektoncd/catalog/tree/master/task/kubectl-deploy-pod/0.1))
+to work with remote k8s cluster.
+
+### Examples
+
+Examples are presented just to describe main idea and can be modified in the
+future to use tasks from [catalog](https://github.com/tektoncd/catalog/tree/master/task).
+
+#### Example 1: Build architecture specific release for Tekton pipeline
```

Comment:
This in-depth detail is really helpful, but it might help to review and approve the TEP if we elide the specifics of how the build happens, and focus on answering 1.) should we do this (definitely), and 2.) is [this] roughly _how_ we should do it, where [this] is purposefully kept high-level.

Reply:
got it, thank you